src/Illuminate/Log/LogManager.php

```diff
@@ -0,0 +1,542 @@
+use Monolog\Handler\StreamHandler;
+use Monolog\Handler\SyslogHandler;
+use Monolog\Formatter\LineFormatter;
+use Monolog\Handler\ErrorLogHandler;
+use Monolog\Handler\HandlerInterface;
+use Illuminate\Log\Events\MessageLogged;
+use Monolog\Handler\RotatingFileHandler;
+
+class LogManager implements LoggerInterface
+{
+    /**
+     * The application instance.
+     *
+     * @var \Illuminate\Foundation\Application
+     */
+    protected $app;
+
+    /**
+     * The array of resolved logs.
+     *
+     * @var array
+     */
+    protected $logs = [];
+
+    /**
+     * The registered custom driver creators.
+     *
+     * @var array
+     */
+    protected $customCreators = [];
+
+    /**
+     * The Log levels.
+     *
+     * @var array
+     */
+    protected $levels = [
+        'debug' => Monolog::DEBUG,
+        'info' => Monolog::INFO,
+        'notice' => Monolog::NOTICE,
+        'warning' => Monolog::WARNING,
+        'error' => Monolog::ERROR,
+        'critical' => Monolog::CRITICAL,
+        'alert' => Monolog::ALERT,
+        'emergency' => Monolog::EMERGENCY,
+    ];
+
+    /**
+     * Create a new Log manager instance.
+     *
+     * @param  \Illuminate\Foundation\Application  $app
+     * @return void
+     */
+    public function __construct($app)
+    {
+        $this->app = $app;
+    }
+
+    /**
+     * Get a log channel instance.
+     *
+     * @param  string  $driver
+     * @return mixed
+     */
+    public function channel($channel = null)
+    {
+        return $this->driver($channel);
+    }
+
+    /**
+     * Get a log driver instance.
+     *
+     * @param  string  $driver
+     * @return mixed
+     */
+    public function driver($driver = null)
+    {
+        return $this->get($driver ?? $this->getDefaultDriver());
+    }
+
+    /**
+     * Attempt to get the log from the local cache.
+     *
+     * @param  string  $name
+     * @return \Psr\Log\LoggerInterface
+     */
+    protected function get($name)
+    {
+        try {
+            return $this->stores[$name] ?? with($this->resolve($name), function ($monolog) use ($name) {
+                return $this->tap($name, new Logger($monolog, $this->app['events']));
+            });
+        } catch (Throwable $e) {
+            return tap($this->createEmergencyLogger(), function ($logger) use ($e) {
+                $logger->emergency('Unable to create configured logger. Using emergency logger.');
+
+                $logger->emergency($e);
+            });
+        }
+    }
+
+    /**
+     * Apply the configured taps for the logger.
+     *
+     * @param  string  $name
+     * @param  \Illuminate\Log\Logger  $logger
+     * @return \Illuminate\Log\Logger
+     */
+    protected function tap($name, Logger $logger)
+    {
+        foreach ($this->configurationFor($name)['tap'] ?? [] as $tap) {
+            list($class, $arguments) = $this->parseTap($tap);
+
+            $this->app->make($class)->__invoke($logger, ...explode(',', $arguments));
```

Comment:
I don't see why we should call `__invoke()` manually here, seems dirty if we're not using `__invoke` for its purpose (treating an object as a method essentially), might as well use another method name then.

Reply:
Do you have a method name suggestion?

Reply:
`handle` would follow the middleware standard. `process` or `execute` could be options too

Reply:
The why I learnt to expect how `handle` works, is that it's arguments were resolved by the Container and injected (think: Console commands and Jobs).